pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1325,31 +1333,44 @@ func TestPriorityQueue_MoveAllToActiveOrBackoffQueue(t *testing.T) {
 	if p, err := q.Pop(); err != nil || p.Pod != hpp1 {
 		t.Errorf("Expected: %v after Pop, but got: %v", hpp1, p.Pod.Name)
 	}
+	unschedulableQueuedPodInfo := q.newQueuedPodInfo(unschedulablePodInfo.Pod, "fooPlugin")
```

Comment:
This entire test is too hard to read (anyone reading has to keep a mental state of all that happened in the previous steps). I wonder if it can be split into different test cases that target specific transitions.
But that would be for a follow up.

Reply:
👍 
I'd like to do that refactor in a follow-up PR. 
I guess somehow we can just merge this test and `TestPriorityQueue_MoveAllToActiveOrBackoffQueueWithQueueingHint`.